ykllvmwrap/src/ykllvmwrap.cc

```diff
@@ -37,15 +52,107 @@ extern "C" char *__yk_llvmwrap_symbolizer_find_code_sym(
     return strdup(LineInfo->FunctionName.c_str());
 }
 
-/// Compiles an IRTrace to executable code in memory.
+// Load an LLVM module from an address.
+std::unique_ptr<Module> load_module(LLVMContext &Context, char *Ptr, size_t Len) {
+    auto Sf = StringRef(Ptr, Len);
+    auto Mb = MemoryBufferRef(Sf, "");
+    SMDiagnostic Error;
+    auto M = parseIR(Mb, Error, Context);
+    if (!M)
+        errx(EXIT_FAILURE, "Can't load module.");
+    return M;
+}
+
+// Compile a module in-memory and return a pointer to its function.
```

Comment:
Is `void *` the best return type here?

Reply:
This used to be `uintptr_t`, but we thought a pointer is nicer. Suggestions on what other type we could use here (we don't really know the type of this function yet).

Reply:
The only other reasonable type would be a function pointer.

Reply:
A function pointer sounds like it would make sense although in fairness I guess we don't know the actual signature i.e. the arguments it takes?

Reply:
Ultimately the pointer has to become a func ptr (with static type) somewhere in order to be called. Currently the test does a cast.

Reply:
There's not a single function pointer type and I don't know what the casting rules are for punning between different function pointer types. It is probably (but I'm not sure) undefined behaviour to cast between different (incorrect) function pointer types.

Reply:
I'd imagined the pointer being varargs accepting only pointers.

Reply:
Sounds to me like we should leave it until we have decided what this function is going to look like (which I assume we'll figure out soon when we make traces to a bit more things, e.g. pass in values for reading/writing).

Reply:
ok

Reply:
I agree with waiting.